pkgs/development/python-modules/robotframework/default.nix

```
@@ -1,15 +1,23 @@
-{ stdenv, fetchPypi, buildPythonPackage }:
+{ stdenv, fetchFromGitHub, buildPythonPackage, tkinter, python }:
 
 buildPythonPackage rec {
   pname = "robotframework";
   version = "3.2.1";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "d693e6d06b17f48669e2a8c4cb6c1f0d56e5f1a74835d18b8ea2118da7bf2d79";
-    extension = "zip";
+  src = fetchFromGitHub {
+    owner = "robotframework";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0drsw2jm6qlx9j15lj71l5bwgmsqa36pcqq61rk02xy6zvsp68dd";
   };
 
+  propagatedBuildInputs = [ tkinter ];
+
+  checkPhase = ''
```

may help with your test issue, you could also try setting it to something like 2010
```suggestion
  checkPhase = ''
    unset SOURCE_DATE_EPOCH
```

---

Unfortunately, neither had any effect... I suppose I could skip these tests?

---

skipping tests which dont work within a sandbox is fine.

not sure what you can do in this situation, but pytest allows you to skip on a test name basis

---

@jonringer, looks like I only needed to skip one test file, so I'm just adding a patch that deletes it. (I don't see a convenient way to skip certain tests with this custom runner they have.) 

I'm also running their "acceptance" tests. The unit tests run in only a few seconds, but the acceptance tests take minutes to complete. Should we include the acceptance tests in checkPhase?

I am actually finding more missing libraries by running their acceptance tests, so it has been beneficial. Is it important to try to avoid making the build process take a long time?

---

Looks like there are several acceptance tests that fail even with all of the dependencies. I think I'll comment them out for now and push my code. Let me know what you think.

---

@jonringer, is there anything else to be done here before I squash? Is the PR in an acceptable state?

---

I think the last two could be squashed, one adds tests, then another makes them pass. Ideally, that would just be one commit